src/ann_solo/spectrum.py

```diff
@@ -214,6 +215,57 @@ def spectrum_to_vector(spectrum: MsmsSpectrum, min_mz: float, max_mz: float,
     return vector
 
 
+def spectrum_to_vector(spectrum: MsmsSpectrum, transformation: ss.csr_matrix,
```

Comment:
**question:** I assume that the sparse vectors need to be converted to dense vectors to be compatible with the Faiss index? Is there a benefit to using `SparseRandomProjection` over `GaussianRandomProjection`?

Reply:
That is correct. Both use random projections, but each has its own advantages.
SparseRandomProjection is more efficient in calculation and requires less memory, so very ideal to very large vectors.
GaussianRandomProjection is not sparse and main advantage as known by the community is its ability to maintain pairwise distance between data points, after transformation. And I think that's what we want to aim for. Let's use a matrix using GaussianRandomProjection for transformation of spectra to low-dim vectors.

Reply:
The [Scikit-Learn documentation](https://scikit-learn.org/stable/modules/random_projection.html) says this:
> Sparse random matrices are an alternative to dense Gaussian random projection matrix that guarantees similar embedding quality while being much more memory efficient and allowing faster computation of the projected data.

Neither this statement nor that the Gaussian random projections should be better at conserving the pairwise distance is immediately obvious to me. Let's evaluate both for our specific context, then we can make an informed decision.